integreat_cms/static/src/js/analytics/hix-widget.ts

```diff
@@ -58,14 +62,71 @@ const setHixLabelState = (state: string) => {
         updateButton.classList.remove("hidden");
     }
 
+    // Hide HIX feedback when the HIX state is not up-to-date
+    if (state !== "updated") {
+        document.getElementById("hix-feedback").classList.add("hidden");
+    } else {
+        document.getElementById("hix-feedback").classList.remove("hidden");
+    }
+
     // Hide the loading spinner
     document.getElementById("hix-loading")?.classList.add("hidden");
 };
 
-const getHixValue = async () => {
+/**
+ * Mapping between category names in the HIX widget and in the Textlab API response
+ */
+const feedbackCategoriesMapping = new Map<string, string>([
+    ["hix-count-nested-sentences", "moreSentencesInClauses"],
+    ["hix-count-long-sentences", "moreSentencesInWords"],
+    ["hix-count-long-words", "moreWordsInLetters"],
+    ["hix-count-passive-voice-sentences", "countPassiveVoiceInSentence"],
+    ["hix-count-infinitive-constructions", "countInfinitiveConstructions"],
+    ["hix-count-nominal-sentences", "countNominalStyle"],
+    ["hix-count-future-tense-sentences", "countFutureTenseInSentence"],
+]);
+
+/**
+ * Display the HIX feedback details
+ */
+const updateHixFeedback = (hixFeedback: string) => {
+    const feedbackCategories: NodeListOf<HTMLElement> = document.querySelectorAll("[hix-feedback-category]");
+    const feedbackJson = JSON.parse(hixFeedback);
+
+    let feedbackCount = 0;
+
+    feedbackCategories.forEach((category) => {
+        const categoryName = category.getAttribute("hix-feedback-category");
+        const feedbackByCategory = feedbackJson[feedbackCategoriesMapping.get(categoryName)];
+
+        if (feedbackByCategory?.length > 0) {
+            category.classList.remove("hidden");
+            const categoryCount: HTMLElement = category.querySelector("span");
+            categoryCount.textContent = feedbackByCategory.length;
+            feedbackCount += 1;
+        } else {
+            category.classList.add("hidden");
+        }
+    });
+
+    const feedbackBlock: HTMLElement = document.getElementById("hix-feedback");
+    if (feedbackCount === 0) {
+        feedbackBlock.classList.add("hidden");
+    } else {
+        feedbackBlock.classList.remove("hidden");
+    }
+};
+
+/**
+ * Request current HIX data
+ * @returns HIX score and HIX feedback
+ */
+const getHixData = async (): Promise<[number, string]> => {
     const updateButton = document.getElementById("btn-update-hix-value");
-    let result;
     const sentContent = getContent().trim();
+
+    let result;
```

Comment:
The result can be assigned directly and then also changed from `let` to `const`. ☺️ 

```
const result = await fetch(updateButton.dataset.url, ...)
    .then((response) => response.json())
    .then((json) => {
        if (json.error || json.score === null) {
            updateHixStateLabel("error");
            return [];
        } else {
            updateHixStateLabel("updated");
            initialContent = sentContent;
            return [json.score, JSON.stringify(json.feedback)];
        }
    });
```

Reply:
Then the result type will be Promise<Response>, while we need Promise<[number, string]>.

Reply:
Sorry, here's an example that should work properly and address the following two points:

- It makes the result a const instead of a let, which removes mutability and makes the code slightly easier to read since the reader doesn't have to think about subsequent places that mutate the variables.
- It also adds error handling on the promise level, which was missing before. This could have led to a case where the server responds with anything except a status code in the range of 200 to 399, and then the client would have failed silently, if I understand correctly.

What are your thoughts on this?

```js
/**
 * Request current HIX data
 * @returns HIX score and HIX feedback
 */
const getHixData = async (): Promise<[number?, string?]> => {
    const updateButton = document.getElementById("btn-update-hix-value");
    const sentContent = getContent().trim();

    try {
        const response = await fetch(updateButton.dataset.url, {
            method: "POST",
            headers: {
                "X-CSRFToken": getCsrfToken(),
            },
            body: JSON.stringify({
                text: sentContent,
            }),
        });

        const json = await response.json();

        if (json.error || json.score === null) {
            updateHixStateLabel("error");
            return [];
        } else {
            updateHixStateLabel("updated");
            initialContent = sentContent;
            return [json.score, JSON.stringify(json.feedback)];
        }
    } catch (error) {
        updateHixStateLabel("error");
        return [];
    }
};
```

Reply:
> This could have led to a case where the server responds with anything except a status code in the range of 200 to 399

Currently the server only responds with 200, even if there is an error.
I believe this could be changed to be more in line with the http protocol, but not within the scope of this PR because we wanted to deliver HIX feedback with a high prio and refactoring existing functionality can be done later.